portal/src/graphql/adminapi/ResetPasswordScreen.tsx

```diff
@@ -0,0 +1,197 @@
+  }
+
+  return violations;
+}
+
+function hasViolation(
+  violationType: ViolationType,
+  violations: ResetPasswordScreenViolation[]
+): boolean {
+  return (
+    violations.filter((violation) => violation.violationType === violationType)
+      .length > 0
+  );
+}
+
+const ResetPasswordForm: React.FC<ResetPasswordFormProps> = function (
+  props: ResetPasswordFormProps
+) {
+  const { appConfig } = props;
+
+  const { renderToString } = useContext(Context);
+
+  const [violations, setViolations] = useState<ResetPasswordScreenViolation[]>(
+    []
+  );
+
+  const passwordPolicy = useMemo(() => {
+    return appConfig?.authenticator?.password?.policy ?? {};
+  }, [appConfig]);
+
+  const { value: newPassword, onChange: onNewPasswordChange } = useTextField(
+    ""
+  );
+  const {
+    value: confirmPassword,
+    onChange: onConfirmPasswordChange,
+  } = useTextField("");
+
+  const screenState = useMemo(
+    () => ({
+      newPassword,
+      confirmPassword,
+    }),
+    [newPassword, confirmPassword]
+  );
+
+  const isFormModified = useMemo(() => {
+    return !deepEqual({ newPassword: "", confirmPassword: "" }, screenState);
+  }, [screenState]);
+
+  const onConfirmClicked = useCallback(() => {
+    const newViolations = validate(screenState, passwordPolicy);
+    setViolations(newViolations);
+    if (newViolations.length > 0) {
+      // return
+    }
+
+    // TODO: integrate mutation
+  }, [screenState, passwordPolicy]);
+
+  const newPasswordErrorMessage = useMemo(() => {
+    if (hasViolation("invalid-password", violations)) {
+      return renderToString("ResetPasswordScreen.error.invalid-password");
+    }
+    return undefined;
+  }, [violations, renderToString]);
+
+  const confirmPasswordErrorMessage = useMemo(() => {
+    if (hasViolation("confirm-password-not-match", violations)) {
+      return renderToString(
+        "ResetPasswordScreen.error.confirm-password-not-match"
+      );
+    }
+    return undefined;
+  }, [violations, renderToString]);
+
+  if (appConfig == null) {
+    return (
+      <Text>
+        <FormattedMessage id="ResetPasswordScreen.error.fetch-password-policy" />
+      </Text>
+    );
+  }
+
+  return (
+    <div className={styles.form}>
+      <NavigationBlockerDialog blockNavigation={isFormModified} />
+      <PasswordField
+        className={styles.newPasswordField}
+        textFieldClassName={styles.passwordField}
+        errorMessage={newPasswordErrorMessage}
+        label={renderToString("ResetPasswordScreen.new-password")}
+        value={newPassword}
+        onChange={onNewPasswordChange}
+        passwordPolicy={passwordPolicy}
+      />
+      <TextField
+        className={cn(styles.passwordField, styles.confirmPasswordField)}
+        label={renderToString("ResetPasswordScreen.confirm-password")}
+        value={confirmPassword}
+        onChange={onConfirmPasswordChange}
+        errorMessage={confirmPasswordErrorMessage}
+      />
```

Comment:
The type should also be `password`, not plaintext.

Reply:
Thanks, forgotten to put type here